### ISSUE TITLE
Upgarde to CKEditor 5 because of EOL of V4

### ISSUE THREAD

**Issue by glu000** [Dolibarr/dolibarr#26112]
### Feature Request

Dolibarr uses CKEditor 4 which reached EOL in June 2023

see https://ckeditor.com/docs/ckeditor4/latest/index.html

### Suggested implementation

Upgrade to CKEditor V5

**Comment by rycks** [comment]
Hello,

> CKEditor 4 and CKEditor 5 are two different products. Here are the most important aspects you need to consider before you migrate.

and

> Because of the differences in features, the data produced with CKEditor 4 may not be compatible with CKEditor 5 which may lead to data loss. Any data that is not supported by features enabled in CKEditor 5 will be removed when loaded into the editor.

Source: https://ckeditor.com/docs/ckeditor5/latest/updating/ckeditor4/migration-from-ckeditor-4.html

**Comment by PlanPc** [comment]
Dolibarr 19.0.2.

Alert message when selecting Full_inline editor at WYSYWYG module options.

**This CKEditor _4.22.1_ version is not secure**. Consider [upgrading to the latest one](https://ckeditor.com/ckeditor-4-support/), **_4.24.0-lts_**.

![image](https://github.com/user-attachments/assets/b764b734-322d-496b-9d57-00ac48c0024f)

**Comment by eldy** [comment]
I think the warning (asking to switch to a paying version) is removed with 19.0.3. Can you check ?

**Comment by PlanPc** [comment]
> I think the warning (asking to switch to a paying version) is removed with 19.0.3. Can you check ?

Soon I will update to 19.0.3, and later report here again after testing.

Thank you very much Laurent.

**Comment by PlanPc** [comment]
The message has changed slightly, but it still asks to upgrade, now to version _4.25.0-lts_., before it asked for _4.24.0.lts_, but presumably, the version of CKEditor that Dolibar has now, is still the same **4.22.1**, just like in Dolibarr 19.0.2, so no chages appreciated.

This warning message is only showed when selecting  **Full_inline** in the WYSIWYG module configuration, dont really know if can be shown in other places inside Dolibarr.

_

> This CKEditor 4.22.1 version is not secure. Consider [upgrading to the latest one](https://ckeditor.com/ckeditor-4-support/), 4.25.0-lts.

_


Donwload last version from [https://ckeditor.com/cke4/release/CKEditor-4.25.0-LTS](https://ckeditor.com/cke4/release/CKEditor-4.25.0-LTS) Where they say: **All editor versions below 4.25.0-lts can no longer be considered as secure!** ⚠️


Direct link to download tar.gz file: [CKEditor-4.25.0-LTS](https://download.cksource.com/CKEditor/CKEditor/CKEditor%204.25.0-lts/ckeditor_4.25.0-lts_standard.tar.gz)

![imagen](https://github.com/user-attachments/assets/d292d16d-a30b-40c0-9823-c07d2a3221ed)

**Comment by aspangaro** [comment]
I can confirm, I found this message displayed in a new migration to v19.0.3.

LTS license available until December 2026.

Maybe the time, we'll have to tackle the integration of CKE5...

**Comment by Jeff-maint** [comment]
I confirm this @lmag can you help us ?
![image](https://github.com/user-attachments/assets/52642b52-f6c4-4365-91b4-4705138cc1bf)

**Comment by lmag** [comment]
Warning still available in 19.0.4

**Comment by lmag** [comment]
Use Full
![image](https://github.com/user-attachments/assets/9b7910a8-be79-440e-ad18-b8469a079984)
Use Full_Inline
![image](https://github.com/user-attachments/assets/35879d11-482e-4903-a98d-a9842ee8beb6)

**Comment by FHenry** [comment]
On the config page , there is this **test** mode "Full_inline",

In Core Dolibarr this mode of display is use only on place :
Live edit in SiteWeb page (@eldy) :
![image](https://github.com/user-attachments/assets/049ee5ca-38c9-46d7-b50f-a3c7f226a212)
But message is "Feature not yet available in this version"

The screenshot of @Jeff-maint come from extarnal module [SubTotal](https://github.com/ATM-Consulting/dolibarr_module_subtotal) the use this display mode (@atm-lena). From my point of view it should not use this mode.

The fact is the CKEditor module with this display mode (and only in this mode) make a HTTP request to https://cke4.ckeditor.com to test the version 
![image](https://github.com/user-attachments/assets/2ac964d1-efaa-4029-ba29-308e2f34f8b0)

So basically all Dolibarr the use subtotal (free text popup), website in live edit, or test this mode in config page send the WAN IP to cke4.ckeditor.com.
For me that's the major point with that issue.

Change version of ckEditor have to be done but this is a huge rework, and will break a lot of external module that use to tweak this component.
